### ISSUE TITLE
Updating record with multiple new values

### ISSUE THREAD

**Issue by Zhomart** [Crecto/crecto#58]
Is there way of updating record with multiple values? 

Example:

```crystal
class User
  scheme do
    field :name, String
    field :age, Int32
  end
end

params = { name: "Crecto", age: 3 }

user = Crecto::Repo.get(User, 1)
user.update(params) # --- how to do it?
Crecto::Repo.update(user)
```

**Comment by fridgerator** [comment]
Currently there is no shortcut method for updating from json, unless there is some helper thats a part of `JSON.mapping` that i'm not aware of

**Comment by Zhomart** [comment]
`JSON.mapping` is fairly simple, it only creates getters, setters, instance variables and initializer.
I think `JSON.mapping` should also create two methods `attributes : Hash`, which is same as `to_query_hash`, and `attributes=(attrs : Hash)` that works like initializer.

**Comment by faultyserver** [comment]
Is this still something that's being considered? If so, I'd like to be able to pass a `params` object _or_ pass naked attributes:

```crystal
u = User.new
params = { name: "tester", age: 21 }
u.update(params)
u.update(name: "tester", age: 21)
```

Crystal allows splatting NamedTuples, so if the bare attributes version is implemented, I think the explicit `params` object could be pretty simple:

```crystal
def update(params : NamedTuple)
  update(**params)
end
```

**Comment by fridgerator** [comment]
I have another branch where I have added a couple of small features to support [another related project](https://github.com/crecto/crecto-admin) where I have implemented an [`update_from_hash` method](https://github.com/Crecto/crecto/blob/crecto_admin/src/crecto/schema.cr#L182).

In this case though its expecting a hash of `Hash(String, String)` type (which comes from `HTTP::Params#to_s`).  I might be able to extend this to allow for a more generic hash argument, and probably an overloaded method for `NamedTuple`

**Comment by faultyserver** [comment]
Nice! That certainly seems like the more common use case. Any idea of when that might get merged?

**Comment by fridgerator** [comment]
Later tonight probably.  I have a couple of tweaks on my computer at home that I haven't pushed yet, but I think its ready to be merged.

**Comment by fridgerator** [comment]
closed by #91
